tests/TurboMqtt.Tests/End2End/TcpMqtt311End2EndSpecs.cs

```diff
@@ -111,14 +111,18 @@ public async Task ShouldTerminateClientAfterMultipleFailedConnectionAttempts()
     [Fact]
     public async Task ShouldFailToConnectToNonExistentServer()
     {
-        var updatedTcpOptions = new MqttClientTcpOptions("localhost", 21884);
+        var updatedTcpOptions = new MqttClientTcpOptions("localhost", 21884)
+        {
+            MaxReconnectAttempts = 0
+        };
         var client = await ClientFactory.CreateTcpClient(DefaultConnectOptions, updatedTcpOptions);
 
         var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
         var connectResult = await client.ConnectAsync(cts.Token);
         connectResult.IsSuccess.Should().BeFalse();
 
         client.IsConnected.Should().BeFalse();
-        client.WhenTerminated.IsCompleted.Should().BeTrue();
+        // ReSharper disable once MethodSupportsCancellation
+        await AwaitAssertAsync(() => client.WhenTerminated.IsCompleted.Should().BeTrue());
```

Comment:
Don't pass in the CTS here - detecting the failed connection might mean `CancellationToken` has already been invoked.